### PR TITLE
Run `twine check` to ensure python package `long_description` is correct

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

## What changes are proposed in this pull request?

Run `twine check` to ensure python package data is correct and detect issues beforehand.

## How is this patch tested?

New GitHub Actions

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: .github/workflows/build-wheel.yml [hunk 1]
```diff
@@ -9,6 +9,9 @@ on:
 jobs:
   build:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        type: ["full", "skinny"]
```
Comment (harupy) [inline .github/workflows/build-wheel.yml:14]: Run this workflow against both full and skinny clients

File: setup.py [hunk 2]
```diff
@@ -108,6 +108,7 @@ def package_files(directory):
     author="Databricks",
     description="MLflow: A Platform for ML Development and Productionization",
     long_description=open("README.rst").read(),
+    long_description_content_type="text/x-rst",
```
Comment (harupy) [inline setup.py:111]: Required this change to pass `twine check --strict`.

File: .github/workflows/build-wheel.yml [hunk 3]
```diff
@@ -28,23 +32,36 @@ jobs:
...
       - name: Build wheel
         id: build-wheel
         run: |
-          pip install wheel
-          python setup.py bdist_wheel
+          if [ "${{ matrix.type }}" = "skinny" ]; then
+            export MLFLOW_SKINNY=true
+          fi
+
+          python setup.py bdist_wheel sdist
```
Comment (harupy) [inline .github/workflows/build-wheel.yml:46]: I just noticed we're missing `sdist`.

File: .github/workflows/test-package-build.yml [hunk 4]
```diff
@@ -19,6 +22,7 @@ jobs:
           python-version: "3.7"
 
       - name: Build UI
+        if: ${{ matrix.type == 'full' }}
         working-directory: mlflow/server/js
         env:
```
Comment (harupy) [inline .github/workflows/test-package-build.yml:27]: Skip UI build for skinny client

File: .github/workflows/test-package-build.yml [hunk 5]
```diff
@@ -1,4 +1,4 @@
-name: Build wheel
+name: Test package build
```
Comment (harupy) [inline .github/workflows/test-package-build.yml:1]: Rename the workflow to more appropriate name.

File: .github/workflows/test-package-build.yml [hunk 6]
```diff
@@ -9,6 +9,9 @@ on:
 jobs:
   build:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        type: ["full", "skinny"]
```
Comment (harupy) [inline .github/workflows/test-package-build.yml:14]: NIt question: How should we call the non-skinnly client 😅?

File: .github/workflows/test-package-build.yml [hunk 7]
```diff
@@ -28,23 +32,36 @@ jobs:
...
           wheel_name=$(basename $wheel_path)
           wheel_size=$(stat -c %s $wheel_path)
           echo "::set-output name=wheel-path::${wheel_path}"
           echo "::set-output name=wheel-name::${wheel_name}"
           echo "::set-output name=wheel-size::${wheel_size}"
 
-      - name: Verify wheel can be installed
+      - name: Run twine check
+        run: |
+          twine check --strict ${{ steps.build-wheel.outputs.wheel-path }}
```
Comment (harupy) [inline .github/workflows/test-package-build.yml:59]: `twine check` just checks whether `long_description` will render correctly on PyPI:

https://twine.readthedocs.io/en/latest/#twine-check

File: .github/workflows/test-package-build.yml [hunk 8]
```diff
@@ -28,23 +32,36 @@ jobs:
...
         id: build-wheel
         run: |
-          pip install wheel
-          python setup.py bdist_wheel
+          if [ "${{ matrix.type }}" = "skinny" ]; then
+            export MLFLOW_SKINNY=true
+          fi
+
+          python setup.py sdist bdist_wheel
+          ls -lh dist
```
Comment (dbczumar) [inline .github/workflows/test-package-build.yml:47]: Do we need this line?

File: .github/workflows/test-package-build.yml (hunk 6 above)
Comment (dbczumar) [reply .github/workflows/test-package-build.yml:14]: full is great! nice naming!

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks @harupy !

File: .github/workflows/test-package-build.yml (hunk 8 above)
Comment (harupy) [reply .github/workflows/test-package-build.yml:47]: This line outputs:

```
-rw-r--r-- 1 runner docker 500K Feb  5 09:26 mlflow-1.13.2.dev0-py3-none-any.whl
-rw-r--r-- 1 runner docker 390K Feb  5 09:26 mlflow-1.13.2.dev0.tar.gz
```

I added this line to confirm dist files are successfully generated + check their file sizes.

**Pull request merged**